### PR TITLE
feat(aws-service-spec): custom build using service-spec-importers

### PR DESCRIPTION
**Pull request by mrgrain**
Updates `aws-service-spec` to run a custom build that is using the public API from `service-spec-importers`.

Also adds a new build step to post the diff between `base` and `head` branch to the PR.

### CODE REVIEW

File: packages/@aws-cdk/aws-service-spec/scripts/build-db.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { FullDatabase } from './full-database';
+import { writeFile } from 'node:fs/promises';
+
+async function main() {
+  const outputFile = 'db.json.gz';
+  const buildReport = 'build-report';
+
+  process.stdout.write('Importing default sources... ');
+  const { db, report } = await new FullDatabase(emptyDatabase(), {
+    // @TODO: Switch this to 'true' once validation passes (this will never happen)
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-service-spec/scripts/build-db.ts:12]: huh?

**Review by kaizencc — APPROVED** [review]
I mean sure. There's a whole lot of other changes here seemingly unrelated to the feat but are just refactoring. Makes it a bit hard to discern which diffs I should be scrutinizing. But the comment on the PR looks fine, and I saw it when it actually posted a real diff as well.

File: packages/@aws-cdk/service-spec-importers/src/cli/import-db.ts [hunk 2]
```diff
@@ -1,55 +1,53 @@
...
-      `Database file ${outputFile} already exists. Please use '--force' to overwrite the existing file.`,
+      `Database file at ${outputFile} already exists. Please use '--force' to overwrite the existing file.`,
     );
   }
 
   const baseDb = await database(options.input);
 
   process.stdout.write('Importing sources... ');
-  const { db, report } = await FullDatabase.buildDatabase(baseDb, {
-    // FIXME: Switch this to 'true' at some point
```
Comment (mrgrain) [inline packages/@aws-cdk/service-spec-importers/src/cli/import-db.ts:38]: @kaizencc this was the original comment. I made it more honest

**Comment by mrgrain** [comment]
> I mean sure. There's a whole lot of other changes here seemingly unrelated to the feat but are just refactoring. Makes it a bit hard to discern which diffs I should be scrutinizing. But the comment on the PR looks fine, and I saw it when it actually posted a real diff as well. 

Will keep it smaller next time 👍

**Pull request merged**